### PR TITLE
School_Library: Preserve data

### PR DESCRIPTION
**Pull request by forLoop94**
### The following changes were implemented:

1. On exit from the app, all the data (books, people, and rentals) is saved into JSON files. One per type of data being stored and named accordingly.
2. On startup of the app, all data from files are loaded. 
3. Cases, when one or more of the files are missing, were also handled.

### CODE REVIEW

File: app.rb
Comment (franclobo) [inline app.rb:1]: - [ ] After creating a people student, it is not saving the data in the JSON file, please fix this issue to preserve the data 👇🏽 

image
image

- [ ] After creating a book it saves the data twice, I suggest removing lies 101 and 102 to fix this issue.

image

image

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @forLoop94 and @ANTOINE1128,

# Required Changes ♻️

so-close-jim-halpert

Good job so far!
You are doing it well 👏🏽 there are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 
- Preserve data with JSON ✔️ 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by forLoop94** [comment]
Issues resolved.
@franclobo, your review is great. Thank you

**Review by addod19 — APPROVED** [review]
## Hi @forLoop94 and @ANTOINE1128, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- [x] Awesome job fixing the bug mentioned in the previous review :rocket: 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**